controllers/kubeadm_control_plane_controller.go

```diff
@@ -191,9 +195,88 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, kcp *clus
 		return ctrl.Result{Requeue: true}
 	}
 
-	logger.Error(errors.New("Not Implemented"), "Not Implemented")
+	// Currently we are not handling upgrade, so treat all owned machines as one for now.
+	// Once we start handling upgrade, we'll need to filter this list and act appropriately
+	numMachines := len(ownedMachines)
+	desiredReplicas := int(*kcp.Spec.Replicas)
```

Comment:
out of scope for this PR but we should add some validation that forces replicas to be >= 0

Reply:
It's already in place, via webhook :)

Reply:
It turns out there is also built in validation on the api-server side when the scale subresource is enabled to force >=0, we just also add additional validation via webhook to ensure that replicas != 0.